tlatools/org.lamport.tlatools/src/util/SequentialResolver.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2025, Oracle and/or its affiliates.
+
+package util;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A resolver that tries several other resolvers in sequence.  The first found result is returned.
+ */
+class SequentialResolver implements Resolver {
+
+    private final List<Resolver> resolvers;
+
+    public SequentialResolver(List<Resolver> resolvers) {
+        this.resolvers = resolvers;
+    }
+
+    @Override
+    public Optional<URL> resolve(String name) throws IOException {
+        for (Resolver resolver : resolvers) {
+            Optional<URL> resolved = resolver.resolve(name);
+            if (resolved.isPresent()) {
+                return resolved;
+            }
+        }
+        return Optional.empty();
```

Comment:
Could this be replaced with something like:
```
resolvers.stream()
.map(resolver -> resolver.resolve())
.flatMap(Optional::stream)
.findFirst();
```

Reply:
Ideally yes, but since `resolve` throws `IOException` it cannot be used cleanly with Java's functional stream API. The for-loop is less elegant for sure, but it does not require exception handling (and hey, it's probably a little more efficient too).